### PR TITLE
fix(plugin_api): wrong metric type enums

### PR DESCRIPTION
**Pull request by mrgian**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:
Fixes wrong metric type enums, which are used to convert plugin metrics to libsinsp metrics.
It also resolves `get_metrics` symbol in plugins. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix(plugin_api): wrong metrics type enums
```

### CODE REVIEW

**Review by incertum — APPROVED** [review]
/approve

Apologies I also totally overlooked this during the last review. Thanks for fixing this.

File: userspace/plugin/plugin_types.h [hunk 1]
```diff
@@ -298,13 +298,13 @@ typedef enum ss_plugin_log_severity
...
-	SS_PLUGIN_METRIC_VALUE_TYPE_D = 5,
-	SS_PLUGIN_METRIC_VALUE_TYPE_F = 6,
-	SS_PLUGIN_METRIC_VALUE_TYPE_I = 7,
+	SS_PLUGIN_METRIC_VALUE_TYPE_U32 = 0,
+	SS_PLUGIN_METRIC_VALUE_TYPE_S32 = 1,
+	SS_PLUGIN_METRIC_VALUE_TYPE_U64 = 2,
+	SS_PLUGIN_METRIC_VALUE_TYPE_S64 = 3,
+	SS_PLUGIN_METRIC_VALUE_TYPE_D = 4,
+	SS_PLUGIN_METRIC_VALUE_TYPE_F = 5,
+	SS_PLUGIN_METRIC_VALUE_TYPE_I = 6,
```
Comment (leogr) [inline userspace/plugin/plugin_types.h:307]: Technically, this is a backward incompatible change, isn't it?

So, unfortunately, I believe we have to bump the `PLUGIN_API_VERSION_MAJOR` number.
cc @jasondellaluce 

/hold a bit for second opinions

Comment (incertum) [reply userspace/plugin/plugin_types.h:307]: agree @leogr we can consider this feature not yet usable in the current release.

Comment (mrgian) [reply userspace/plugin/plugin_types.h:307]: Yet another option:
Forcing the enum values in `metrics_v2.h` (https://github.com/falcosecurity/libs/blob/master/userspace/libscap/metrics_v2.h) to match the enums in `plugin_types.h`
This should work given that these enums are only casted here https://github.com/falcosecurity/libs/blob/master/userspace/libsinsp/plugin.cpp#L947-L948

Comment (FedeDP) [reply userspace/plugin/plugin_types.h:307]: Agree with Leo, since it was unusable in 0.17.1, we can consider a minor bump.

**Comment by mrgian** [comment]
Bumped `PLUGIN_API_VERSION_MINOR` :)

File: userspace/libsinsp/metrics_collector.cpp [hunk 2]
```diff
@@ -783,6 +770,19 @@ void libs_metrics_collector::snapshot()
 			}
 		}
 	}
+
+	/*
+	* plugins metrics
+	*/
+
+	if(m_metrics_flags & METRICS_V2_PLUGINS)
```
Comment (FedeDP) [inline userspace/libsinsp/metrics_collector.cpp:778]: Why was this moved?

Comment (FedeDP) [reply userspace/libsinsp/metrics_collector.cpp:778]: Just curious :)

Comment (mrgian) [reply userspace/libsinsp/metrics_collector.cpp:778]: If we insert plugin metrics first in the vector, they then get overwritten by scap/sinsp metrics.
I didn't look closely at it, but the cause may be: https://github.com/falcosecurity/libs/blob/master/userspace/libsinsp/metrics_collector.cpp#L487

**Review by jasondellaluce — APPROVED** [review]
